src/main/java/io/zold/api/RtTransaction.java

```diff
@@ -75,12 +84,38 @@ public long amount() {
         throw new UnsupportedOperationException("amount() not yet implemented");
     }
 
-    // @todo #15:30min Implement prefix() by parsing the string representation
-    //  of transaction according to the pattern, described in the white
-    //  paper. Replace relevant test case with actual tests.
     @Override
-    public String prefix() {
-        throw new UnsupportedOperationException("prefix() not yet implemented");
+    public String prefix() throws Exception {
```

Comment:
@paulodamaso not a good idea to declare throwing `Exception`. Users of this class will be forced to handle this which is not necessary.

Reply:
@carlosmiranda I've chosen to use `Exception` because I've had to declare this throws in `Transaction` interface. The `IllegalArgumentException` is used by ` RtTransaction`, but other implementations of `Transaction` would throw other types of exceptions, so I can't tie the throws in `Transaction` interface to a exception thown by some implementation.

Reply:
@paulodamaso I agree with @carlosmiranda: let's not throw `Exception` here. Let's wrap it in an `IOException`.